gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbDataTable.java

```diff
@@ -0,0 +1,49 @@
+package org.sakaiproject.gradebookng.tool.component;
+
+import org.apache.wicket.extensions.markup.html.repeater.data.table.AbstractColumn;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.DataTable;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.IColumn;
+import org.apache.wicket.markup.ComponentTag;
+import org.apache.wicket.markup.repeater.data.IDataProvider;
+import org.apache.wicket.markup.repeater.Item;
+import org.apache.wicket.model.IModel;
+
+import java.util.List;
+import java.util.Map;
+
+public class GbDataTable<T, S> extends DataTable<T, S> {
+
+	IModel<Map<String, Object>> gradebookData;
+
```

Comment:
It would have to hit the service. this is only called once on construction though right? Does the GradebookPage even need the assignments/category definitions if all of the work is occuring in here? Could they just be passed in?